SimeonInstaller.ps1

```diff
@@ -512,6 +515,124 @@ CRLFOption=CRLFAlways
             } | ConvertTo-Json) | Out-Null
     }
 
+    <#
+    .SYNOPSIS
+        Invokes the given command block with retries on failure
+    #>
+    function Invoke-WithRetry {
+        [CmdletBinding()]
+        param(
+            [ValidateNotNullOrEmpty()]
+            [scriptblock]$ScriptBlock,
+            [int]$MaxRetryCount = 5,
+            [int]$DelaySeconds = 10
+        )
+
+        $retryCount = 0
+        while ($true) {
+            try {
+                . $ScriptBlock
+                return
+            }
+            catch {
+                $ex = $_.Exception
+                if ($ex -is [System.AggregateException] -and $ex.InnerExceptions.Count -eq 1 -and $ex.InnerExceptions[0].Message) {
+                    $ex = $ex.InnerExceptions[0]
+                }
+                $operation = $ScriptBlock -split '\n' | % { $_.Trim() } | ? { $_ } | Select -First 1
+                Write-Warning "Operation '$operation' failed. Trying again in $DelaySeconds seconds - Exception: $($ex.Message)"
+                Start-Sleep -Seconds $DelaySeconds
+                $retryCount++
+                if ($retryCount -ge $MaxRetryCount) {
+                    throw
+                }
+            }
+        }
+    }
+
+    <#
+    .SYNOPSIS
+        Sets Azure DevOps project permissions
+    #>
+    function Set-AzureDevOpsAccessControlEntry {
+        param(
+            [ValidateNotNullOrEmpty()]
+            [string]$Organization,
+            [ValidateNotNullOrEmpty()]
+            [string]$ProjectId,
+            [ValidateNotNullOrEmpty()]
+            [string]$SubjectGroupPrincipalName,
+            [ValidateNotNullOrEmpty()]
+            [string]$PermissionDescription,
+            [ValidateNotNullOrEmpty()]
+            [int]$PermissionNumber
+        )
+        $groups = (Invoke-WithRetry { Invoke-RestMethod -Header $authenicationHeader -Uri "https://vssps.dev.azure.com/$Organization/_apis/graph/groups?api-version=6.1-preview.1" -Method Get }).value
```

Comment:
you shouldn't assume this variable is defined in a parent scope - each function should get its own auth header